openssl/src/macros.rs

```diff
@@ -144,20 +144,16 @@ macro_rules! foreign_type_and_impl_send_sync {
     )
         => {
             ::foreign_types::foreign_type! {
-                $(#[$impl_attr])*
-                type CType = $ctype;
-                fn drop = $drop;
-                $(fn clone = $clone;)*
                 $(#[$owned_attr])*
-                pub struct $owned;
                 $(#[$borrowed_attr])*
-                pub struct $borrowed;
-            }
+                pub unsafe type $owned : Sync + Send {
 
-            unsafe impl Send for $owned{}
-            unsafe impl Send for $borrowed{}
-            unsafe impl Sync for $owned{}
-            unsafe impl Sync for $borrowed{}
+                    $(#[$impl_attr])*
```

Comment:
I don't know whether this should be put at `pub unsafe type $owned : Sync + Send {`